libraries/default_handler.rb

```diff
@@ -58,12 +58,6 @@ def configure_from_source_package(action_name, local_path = nil)
         options new_resource.options
         source local_path || new_resource.package_source
         timeout new_resource.timeout if new_resource.timeout
-        provider value_for_platform_family(
```

Comment:
This is required because this resource always uses the source property. provider value is required when specifying source. Chef build infrastructure also still needs to support RHEL 5. This breaks all the things.

Reply:
The correct thing to do is to call the correct package resource then. If we need dpkg then dpkg_package, if RPM then rpm_package, etc.

Reply:
That's essential what this does. Instead of creating duplicate resource blocks which have to be condition checked or guarded we just reuse the package attributes and always call the same resource block. This feels like a cleaner solution to me. I'm not tied to anything so I'm cool with getting more opinions on this.

Reply:
We've been told that specifying a provider on the package resource breaks many underlying assumptions that package provider makes and does not result in what you think it results in under the hood. That's why we created the Foodcritic rule to find these sorts of things. I'd point to the Foodcritic docs for it, but I'm a bit behind writing them.

Reply:
OK. If that's the recommended approach I'm fine with that. However, this PR would need to reflect those changes, and not just delete the offending code. chef-ingredient will not function without it.